lighthouse-core/gather/driver.js

```diff
@@ -490,14 +488,53 @@ class Driver {
   }
 
   /**
-   * Navigate to the given URL. Use of this method directly isn't advised: if
+   * Set up listener for network quiet events and URL redirects. URL in options
+   * object will be updated to the post-redirected URL.
+   * @param {{url: string}} options
+   * @return {!Promise}
+   * @private
+   */
+  _beginNetworkMonitoring(options) {
+    return new Promise((resolve, reject) => {
+      this._networkStatus = new NetworkRecorder([], this);
+      this.enableUrlUpdateIfRedirected(options);
+
+      this.sendCommand('Network.enable').then(resolve, reject);
```

Comment:
can't we just return the result of `this.sendCommand` instead of wrapping in another promise?